Added commands for scripted double wild battles

Pyredrid opened the pull request:
## Description
This is a feature PR that adds two scripting commands, setdoublewildbattle and dodoublewildbattle. They function similarly to setwildbattle and dowildbattle, but allow for double battles.

The current battle engine does not support starting a double battle when the player has a single pokemon however, and this PR does not address that.  As such, that edgecase does still exist right now and developers should be aware of it.

## **Discord contact info**
Pyredrid#6298

ghoulslash commented on src/scrcmd.c:
```diff
@@ -2302,3 +2302,23 @@ bool8 ScrCmd_warpsootopolislegend(struct ScriptContext *ctx)
+    return FALSE;
+}
+
+bool8 ScrCmd_dodoublewildbattle(struct ScriptContext *ctx)
```
I think this should have some protection, eg:
```
bool8 ScrCmd_dodoublewildbattle(struct ScriptContext *ctx)
{
    if (CountPartyAliveNonEggMons() > 1)
    {
        BattleSetup_StartScriptedDoubleWildBattle();
        ScriptContext1_Stop();
        return TRUE;
    }
}
```
or default to a single wild battle instead

Pyredrid replied:
While normally I'd agree with something like that, its not the way double battles with a single party member is handled elsewhere.

Using the gym battle for Tate & Liza as an example, there's a check for how many party members the player has before triggering the battle in the script, not as part of the script command.

In any case, I'll go make an issue for how to handle the double battles with a single party member edgecase, since later generations let those battles start as intended.

ultima-soul replied:
My above suggestion is relevant to this topic.

ultima-soul commented on asm/macros/event.inc:
```diff
@@ -1778,3 +1778,21 @@
+
+	@ Prepares to start a double wild battle against a species1 at Level level1 holding item1 and a species2 at Level level2 holding item2. 
+	@ Running this command will not affect normal wild battles. You start the prepared battle with dodoublewildbattle.
+	@ BUG: If the player only has one pokemon, the scripted double battle has undefined behaviours...
```
I'd remove this comment and maybe mention seeing the pokeemerald Expansion wiki for a tutorial on how to use it.

ExpoSeed replied:
i don't see why we can't do both. mention that you should verify the player has 2+ mons and also plug the wiki

Pyredrid replied:
I like Expo's idea  :v

Pyredrid replied:
Expo's idea of both mentioning the bug and linking to a wiki page makes the most sense so far.  For now though, should we pull and I'll make another PR once the wiki page is up or should the wiki page be made first before and I'll make another commit on this PR?  Also, can both @ultima-soul and @ghoulslash comment on the idea of having mentions of both so I know what to do?

ultima-soul replied:
Oops. I thought the needing 2 Pokemon for a proper battle was mentioned before the BUG line when it wasn't at all. d:

But yeah, I think rather than highlighting it as a BUG, just mentioning that you should have 2 mons for normal behaviour should be fine and then updating later with a wiki link to a tutorial on how to script a wild double battle properly.

Pyre mentioned different behaviour for handling similar to future gens, but this should be made an issue rather than being commented as a BUG so that we can look into possibilities and research it more.

Pyredrid replied:
Already made an issue #537 for it a bit ago, the comment was mostly there so if someone finds the bug in the meantime, its already documented as being known.

ultima-soul reviewed (changes requested):
Functionality-wise it's good.

ShinyDragonHunter commented:
Using `dowildbattle` along with `setdoublewildbattle` actually puts the game into a single battle but when the first Pokémon is defeated, it actually sends out the next one straight after, also giving the player to switch Pokémon if the option is set to `Shift`.

ShinyDragonHunter commented:
No. I’m wrong, it doesn’t give the player the option to switch, whether they’ve got the option set to `shift` or not.

Pyredrid commented:
I've updated how this feature works.  Its now a set of optional parameters on the vanilla `setwildencounter` script command instead of two new commands that take up space for any future devs.  I've also changed the comments on the script command to simply state that starting a double battle with one party pokemon is buggy instead of undefined.

When we have a wiki page set up for how to use this feature properly or if we resolve issue #537, I'll make another PR to change the comments.
